### PR TITLE
adb-commands module refactoring

### PR DESCRIPTION
**Pull request by vmaxim**

### CODE REVIEW

File: lib/tools/adb-commands.js [hunk 1]
```diff
@@ -800,11 +798,19 @@ methods.setHttpProxy = async function (proxyHost, proxyPort) {
...
-  await this.shell(['settings', 'put', 'system', 'global_http_proxy_host', proxyHost]);
-  await this.shell(['settings', 'put', 'system', 'global_http_proxy_port', proxyPort]);
+  await this.setSetting('global', 'http_proxy', proxy);
+  await this.setSetting('secure', 'http_proxy', proxy);
+  await this.setSetting('system', 'http_proxy', proxy);
+  await this.setSetting('system', 'global_http_proxy_host', proxyHost);
+  await this.setSetting('system', 'global_http_proxy_port', proxyPort);
+};
+
+methods.setSetting = async function (namespace, setting, value) {
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: Please add docstrings  to these both methods

File: lib/tools/adb-commands.js [hunk 2]
```diff
@@ -800,11 +798,19 @@ methods.setHttpProxy = async function (proxyHost, proxyPort) {
...
+  await this.shell(['settings', 'put', namespace, setting, value]);
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: return might be also useful here

**Comment by imurchie** [comment]
Linting is broken. Please run `npm run lint` locally.

File: lib/tools/adb-commands.js [hunk 3]
```diff
@@ -805,10 +805,23 @@ methods.setHttpProxy = async function (proxyHost, proxyPort) {
...
+*   setting: property name
+*   value: property value
+* https://developer.android.com/reference/android/provider/Settings.html
+*/
 methods.setSetting = async function (namespace, setting, value) {
-  await this.shell(['settings', 'put', namespace, setting, value]);
+  return await this.shell(['settings', 'put', namespace, setting, value]);
 };
 
+/*
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: Please rebase your branch with the recent master to see docstring examples. We do follow [JSDoc tutorial](http://usejsdoc.org/) while creating docstrings.

Comment (vmaxim) [reply lib/tools/adb-commands.js]: oops...

File: test/functional/adb-commands-e2e-specs.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { apiLevel, platformVersion, MOCHA_TIMEOUT } from './setup';
 import { fs, mkdirp } from 'appium-support';
 import temp from 'temp';
 
-
+const should = chai.should();
```
Comment (jlipps) [inline test/functional/adb-commands-e2e-specs.js]: the way that i usually get around lint errors here is simply:

```js
chai.should();
```

i.e., you don't need to assign it to a variable for it to do the thing you need it to do.

**Comment by imurchie** [comment]
Thanks! Functional tests pass locally, and fix for Travis is in master, so merging.

**Pull request merged**